src/main/java/org/broadinstitute/hellbender/engine/FeatureDataSource.java

```diff
@@ -248,8 +267,31 @@ public FeatureDataSource(final FeatureInput<T> featureInput, final int queryLook
             this.hasIndex = false;
             this.supportsRandomAccess = true;
         } else if (featureReader instanceof AbstractFeatureReader) {
+
             this.hasIndex = ((AbstractFeatureReader<T, ?>) featureReader).hasIndex();
             this.supportsRandomAccess = hasIndex;
+
+            // Check that our index is not out-of-date:
+            if( hasIndex ) {
+                final Path featureFilePath = IOUtils.getPath( featureInput.getFeaturePath() );
+
+                if ( !Files.exists(featureFilePath) ) {
+                    throw new GATKException( "Index file should exist, but unable to find it!  Unable to check if index is out of date!" );
+                }
+                else {
+                    // Get our index file:
+                    // Try tribble first, if it's not there, we try tabix after:
+                    Path indexFilePath = Tribble.indexPath(featureFilePath);
```

Comment:
There should be a "getIndex" method somewhere public that does this, but that's just a gripe with the world, not something you have to do here...

Reply:
Good news, there is!  IndexUtils.loadIndex seems to do this, but it takes a file instead of a path :(.  Lets add a new overload that takes a path there.

Reply:
Sounds good.  I made a getIndexPath method in IndexUtils (I don't need to actually load in the index to check the modification times).